lib/Analysis/AxisInfo.cpp

```diff
@@ -68,6 +68,9 @@ AxisInfo AxisInfo::getPessimisticValueState(Value value) {
   auto rank = 1;
   if (TensorType ty = value.getType().dyn_cast<TensorType>())
     rank = ty.getRank();
+  if (triton::PointerType ty = value.getType().dyn_cast<triton::PointerType>())
+    if (TensorType elemTy = ty.getPointeeType().dyn_cast<TensorType>())
+      rank = elemTy.getRank();
```

Comment:
There's a function called `getElementTypeOfTensorPointerType`. Maybe we can use it?

Reply:
but that return the underlying element type so I cannot get the rank from it.